.github/workflows/esti.yaml

```diff
@@ -1021,3 +1022,128 @@ jobs:
           custom-arguments: './clients/python-wrapper/tests/integration -rP'
           click-to-expand: true
           report-title: 'Python Wrapper System Tests Report'
+
```

Comment:
Wondering why is this part of the esti workflow and not in a separate one?

Reply:
I discussed this with @nopcoder as well. I think it should be in a separate file. However, that would require either extracting the Docker image build to a separate workflow and have that trigger the esti and E2E workflows, which I don't like at all; Or building another Docker image in the E2E workflow, and that's not great either, because the Docker build and publish job is one of the longest jobs in the workflow. Another option is to output the image details from the esti workflow and have it trigger the E2E workflow, but then the E2E workflow will only start once the esti workflow is done, which will extend the overall running time of the entire pipeline.
In any case, we agreed that doing a big refactor right now is a defocus from the intention of this PR and we can always refactor later. WDYT?

btw: if you have an idea of how we can improve this elegantly, I'd be happy to discuss.

Reply:
The only elegant idea I'm thinking of is creating an action for the docker image and using it in both workflows.
As long as there's an issue for it, I'm fine deferring

Reply:
Done. #7162